src/main/java/com/tenx/ms/retail/order/rest/OrderController.java

```diff
@@ -0,0 +1,41 @@
+package com.tenx.ms.retail.order.rest;
+
+import com.tenx.ms.commons.exception.RESTCheckedException;
+import com.tenx.ms.commons.rest.RestConstants;
+import com.tenx.ms.retail.order.rest.dto.Order;
+import com.tenx.ms.retail.order.rest.dto.OrderResult;
+import com.tenx.ms.retail.order.service.OrderService;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+
+@Api("Api for orders")
+@RestController
+@RequestMapping(value = RestConstants.VERSION_ONE + "/orders")
+public class OrderController {
+    @Autowired
+    private OrderService orderService;
+
+    @ApiOperation("Create a new Order")
+    @ApiResponses(value = {@ApiResponse(code = 201, message = "Order created successfully"), @ApiResponse(code = 412, message = "Precondition failed, Validation failed"),
+        @ApiResponse(code = 417, message = "Expectation failed, Validation failed"), @ApiResponse(code = 500, message = "Internal service error")})
+    @RequestMapping(method = RequestMethod.POST, consumes = {"application/json"}, path = "/{storeId:\\d+}")
+    @ResponseBody
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
```

Comment:
All these PreAuthroize annotations are tied to security, which is not used in this project. In order to use it, you'll have to extend TokenAwareController in this controller class and have oauth key setup in yml file, etc. You can look at other existing microservices for details, however these security related annotations can be removed from this project.